File: bot/skills/pr.py
import logging

from telegram import Update, User
from telegram.ext import Application, ContextTypes

from handlers import ChatCommandHandler

logger = logging.getLogger(__name__)

MSG = (
    "Would you like to make PR for this?\n"
    "You can start by forking me at https://github.com/vldc-hq/vldc-bot\n"
    "💪😎"
)


def add_pr(app: Application, handlers_group: int):
    logger.info("registering PR handler")
    app.add_handler(ChatCommandHandler("pr", _pr), group=handlers_group)


async def _pr(update: Update, context: ContextTypes.DEFAULT_TYPE):
    if update.message is None:
        return
    user: User | None = (
        update.message.reply_to_message.from_user
        if update.message.reply_to_message
        else None
    )
    msg = f"@{user.username} " + MSG if user else MSG
    await context.bot.send_message(update.effective_chat.id, msg)

File: internal/skill/pr.go
package skill

import (
	"context"
	"fmt"

	"github.com/go-telegram/bot"
	"github.com/go-telegram/bot/models"

	appbot "github.com/vldc-hq/vldc-bot/internal/bot"
)

const prMessage = "Would you like to make PR for this?\n" +
	"You can start by forking me at https://github.com/vldc-hq/vldc-bot\n" +
	"💪😎"

func PrSkill() appbot.Skill {
	return appbot.Skill{
		Name: "pr",
		Hint: "got sk1lzz?",
		Register: func(b *bot.Bot, _ *appbot.Deps) {
			b.RegisterHandler(bot.HandlerTypeMessageText, "/pr", bot.MatchTypePrefix, func(ctx context.Context, b *bot.Bot, update *models.Update) {
				if update.Message == nil {
					return
				}
				msg := update.Message

				text := prMessage
				if msg.ReplyToMessage != nil && msg.ReplyToMessage.From != nil && msg.ReplyToMessage.From.Username != "" {
					text = fmt.Sprintf("@%s %s", msg.ReplyToMessage.From.Username, prMessage)
				}

				_, _ = b.SendMessage(ctx, &bot.SendMessageParams{
					ChatID: msg.Chat.ID,
					Text:   text,
				})
			})
		},
	}
}
